web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx

```diff
@@ -350,6 +376,121 @@ const CreateAccessRole = ({
   );
 };
 
+type TableData = {
+  items: SecurityGroup[];
+  nextToken?: string;
+  fetchStatus: FetchStatus;
+};
+
+const SelectSecurityGroups = ({
+  selectedSecurityGroups,
+  setSelectedSecurityGroups,
+  dbMeta,
+  emitErrorEvent,
+}: {
+  selectedSecurityGroups: string[];
+  setSelectedSecurityGroups(sg: string[]): void;
+  dbMeta: DbMeta;
+  emitErrorEvent(err: string): void;
+}) => {
+  const [sgTableData, setSgTableData] = useState<TableData>({
+    items: [],
+    nextToken: '',
+    fetchStatus: 'disabled',
+  });
+
+  const {
+    attempt: fetchSecurityGroupsAttempt,
+    setAttempt: setFetchSecurityGroupsAttempt,
+  } = useAttempt('processing');
+
+  function onSelectSecurityGroup(
+    sg: SecurityGroup,
+    e: React.ChangeEvent<HTMLInputElement>
+  ) {
+    if (e.target.checked) {
+      return setSelectedSecurityGroups([...selectedSecurityGroups, sg.id]);
+    } else {
+      setSelectedSecurityGroups(
+        selectedSecurityGroups.filter(id => id !== sg.id)
+      );
+    }
+  }
+
+  async function fetchSecurityGroups() {
+    const integration = dbMeta.integration;
+    const selectedDb = dbMeta.selectedAwsRdsDb;
+
+    setFetchSecurityGroupsAttempt({ status: 'processing' });
+    try {
+      const { securityGroups, nextToken } =
+        await integrationService.fetchSecurityGroups(integration.name, {
+          vpcId: selectedDb.vpcId,
+          region: selectedDb.region,
+          nextToken: sgTableData.nextToken,
+        });
+
+      setFetchSecurityGroupsAttempt({ status: 'success' });
+      setSgTableData({
+        nextToken: nextToken,
+        fetchStatus: nextToken ? '' : 'disabled',
+        items: [...sgTableData.items, ...securityGroups],
+      });
```

Comment:
Couldn't we only have these lines in `fetchSecurityGroups()`? Then we would pass this function to `run()` from `useAttempt`. Thanks to that, we wouldn't have to set the attempt statuses manually.